### PR TITLE
[spi_device,dv] Split sim config into separate 1r1w and 2p versions

### PR DESCRIPTION
**Pull request by rswarbrick**
This follows the scheme used for the aes block (where we select between masked/unmasked implementations).

The "implementation" is that we pass a define (SRAM_TYPE). This essentially does the same job as the SRAM_TYPE parameter that was in spi_device_env_pkg.

But now we have a pair of _sim_cfg.hjson files that implement the two modes. This is equivalent to aes_masked_sim_cfg.hjson / aes_unmasked_sim_cfg.hjson.

@msfschaffner: Is this what you were imagining? The notes from the meeting on Friday talked about noting this in the testplan, but the AES testplan (from which I nicked the design) doesn't get split up like that. If there is a better way, I'm happy to rejig it.

### CODE REVIEW

File: hw/ip/spi_device/dv/base_sim_cfg.hjson
Comment (msfschaffner) [inline hw/ip/spi_device/dv/base_sim_cfg.hjson:57]: Can you remove the build modes and pipe the plusdefine in via buildopts as done for lc_ctrl?

https://github.com/lowRISC/opentitan/blob/7e196e6f072791aa17817b7691aed6b258399eeb/hw/ip/lc_ctrl/dv/lc_ctrl_volatile_unlock_disabled_sim_cfg.hjson#L13-L14

I ran into issues collecting coverage with this variant previously.

**Review by msfschaffner — APPROVED** [review]
Yes that's what I meant!

**Review by a-will — COMMENTED** [review]
Seems good to me, but I'll duck out again and let you experts work. I only dabble here and am unfamiliar with the spectrum of choices and their effects. :smile:

File: hw/ip/spi_device/dv/base_sim_cfg.hjson
Comment (rswarbrick) [reply hw/ip/spi_device/dv/base_sim_cfg.hjson]: Yep, happily. I've just pushed up a version that I *think* should work. Leaving this for you to glance at before it gets merged but please feel free to hit merge if it looks sensible to you.

**Comment by msfschaffner** [comment]
LGTM, although it seems like it needs another `make -C hw`...

**Review by msfschaffner — APPROVED** [review]

**Comment by rswarbrick** [comment]
Oh! I think that rebasing should suffice (so that I pick up 7771831eae41ccf0e2c2a0cdac5bef933bb6d62e). I've just done so. I'll merge if the quick lint check passes.

**Pull request merged**
